core/src/main/scala/io/delta/tables/DeltaTableBuilder.scala

```diff
@@ -287,6 +290,27 @@ class DeltaTableBuilder private[tables](
     this
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Specify whether table properties should be automatically converted to lower case.
+   * This allow to respect the past behaviour of the DeltaTableBuilder compatibility. See
+   * {@link io.delta.tables.DeltaTableBuilder.property()}
+   *
+   * @param forceTablePropertyLowerCase the value of the flag.
+   *                                    If true, table properties will be converted to lowercase
+   * @since 1.3.0
+   */
+  @Evolving
+  def forceTablePropertyLowerCase(forceTablePropertyLowerCase: Boolean): DeltaTableBuilder = {
```

Comment:
Could you add a DeltaSQLConf instead? We don't want to add a new API for a legacy case.